common/short-hex-str/src/lib.rs

```diff
@@ -87,9 +88,10 @@ fn byte2hex(byte: u8) -> (u8, u8) {
 }
 
 /// Hex encode a byte slice into the destination byte slice.
+#[allow(clippy::integer_arithmetic)]
 #[inline(always)]
 fn hex_encode(src: &[u8], dst: &mut [u8]) {
-    debug_checked_precondition!(dst.len() == 2 * src.len());
+    debug_checked_precondition!(dst.len() == src.len().saturating_mul(2));
```

Comment:
this is a debug-only assertion, I would actually _prefer_ it to panic on wrap. so leaving it as `2 * src.len()` is best. or `src.len().overflowing_mul(2).unwrap()` lol